### DISCUSSION TITLE
Documentation preview builds on CI

### DISCUSSION THREAD

**Discussion by bertiewooster** [rdkit/rdkit#6849] [Ideas]
I noticed that code in docstrings was not formatted correctly in the documentation for the [MolsMatrixToGridImage](https://www.rdkit.org/docs/source/rdkit.Chem.Draw.html#rdkit.Chem.Draw.MolsMatrixToGridImage) code I submitted (scroll down to EXAMPLES), then discovered that two other docstrings had similar issues, so I submitted PR #6848. There are likely more issues throughout the documentation.

To find these issues before they get published upon a release, it would be helpful to have CI build the docs for pull requests. It might also be a good idea to have an in-development version of the docs so contributors can check for formatting problems before a release. As an example, the [SymPy documentation](https://docs.sympy.org/dev/index.html) that I contributed to had both.

Are documentation preview build on CI something we'd consider, and feasible?

**Comment by greglandrum** [comment]
Yeah, it would be cool to have something that builds the docs after something is merged to master and then copies them somewhere publicly visible. Maybe there is some way to set up a GitHub action that copies to ReadTheDocs.

As for having it as part of the CI builds: we can easily produce a tar ball with the docs, and I think that there is some way to make that artifact available as long as the CI build is kept around.

Pull requests are welcome. ;-)

**Reply by greglandrum** [reply]
Hmm, looks like the preview may be easy: https://github.com/readthedocs/actions/blob/v1/preview/README.md

I will try to make some time for this soon
